aten/src/ATen/native/quantized/cpu/qconcat.cpp

```diff
@@ -8,35 +8,57 @@
 
 namespace at { namespace native {
 namespace {
+
+/* Quantized concatenation.
+
+Note: This function uses a dequantization
+*/
+template <bool ReLUFused>
+Tensor quantized_cat(const std::vector<Tensor>& qxs, int64_t axis,
+                     double scale, int64_t zero_point) {
+  const auto x_dtype = qxs[0].scalar_type();
+  std::vector<Tensor> xs;
+  xs.reserve(qxs.size());
+  for (const auto& qx: qxs) {
+    TORCH_CHECK(x_dtype == qx.scalar_type(),
+                 "All dtypes must be the same.");
+    xs.push_back(qx.dequantize());
+  }
+  const Tensor y = at::cat(xs, axis);
+  Tensor qy;
+  AT_DISPATCH_QINT_TYPES(x_dtype, "qcat", [&]() {
+    qy = at::quantize_linear(y, scale, zero_point, SCALAR_TYPE);
+    if (ReLUFused) {
+      auto iter = TensorIterator::unary_op(qy, qy);
+      cpu_kernel(*iter, [&](scalar_t value) -> scalar_t {
+        return scalar_t(std::max<underlying_t>(value.val_, zero_point));
+      });
+    }
+  });
+  return qy;
+}
+
 template <bool ReLUFused = false>
 class QCat final : public torch::OperatorKernel {
  public:
   Tensor operator()(const std::vector<Tensor>& qxs, int64_t axis,
                     c10::optional<double> scale,
                     c10::optional<int64_t> zero_point) {
-    const auto x_dtype = qxs[0].scalar_type();
     double _scale = scale ? *scale : qxs[0].q_scale();
     int64_t _zero_point = zero_point ? *zero_point : qxs[0].q_zero_point();
```

Comment:
Not sure about the best practice to use c10::optional. Should we use ```zero_point.has_value()``` and ```zero_point.value()``` instead of the raw pointer way?